app/models/miq_request_task/state_machine.rb

```diff
@@ -21,29 +21,15 @@ def signal(phase)
     begin
       send(phase)
     rescue => err
-      if [:finish, :provision_error].include?(phase.to_sym)
-        _log.error("[#{err.message}] encountered during [#{phase}]")
-        _log.log_backtrace(err)
+      case phase
+      when :finish
+        $log.error("[#{err.class}: #{err.message}] encountered during [#{phase}]")
+        $log.log_backtrace(err)
+      when :provision_error
+        update_and_notify_parent(:state => "finished", :status => "Error", :message => err.message)
+        signal(:finish)
       else
-        phase_context.delete(:error_message)
-
-        # HACK: Psych is unable to deserialize an Exception that contains a
-        #   NameError::message as its message.  When deserialized, it will raise
-        #   "TypeError: allocator undefined for NameError::message".  Calling
-        #   .message will dereference it and turn it into a String which is
-        #   serializable.  For more information about NameError::message, see
-        #   http://www.carboni.ca/blog/p/Ruby-Did-You-Know-That-5-NameErrormessage
-        #
-        #   In addition, backtrace information is not available, because it is
-        #   not an instance variable, but is instead dynamically generated.
-        #
-        #   As such, we store the exception's components separately in the
-        #   phase context.
-        phase_context[:exception_class]     = err.class.name
-        phase_context[:exception_message]   = err.message
-        phase_context[:exception_backtrace] = err.backtrace
-        phase_context[:error_phase]         = phase
-        signal :provision_error
```

Comment:
I'm confused... Did you just completely eliminate the storing of exceptions in the db altogether?

Additionally, there may have been callers (perhaps via api or automate) that were expecting these calls in the phase context.

Reply:
Based on the comments, it looked to me like the reason for setting the `phase_context` explicitly like that was no longer necessary and that the `save` method already did it. Specifically, I was looking at this comment:

https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_request_task/state_machine.rb#L75

Are you saying I should explicitly restore these four lines?

```
phase_context[:exception_class]     = err.class.name
phase_context[:exception_message]   = err.message
phase_context[:exception_backtrace] = err.backtrace
phase_context[:error_phase]         = phase
```

Reply:
>  it looked to me like the reason for setting the phase_context explicitly like that was no longer necessary and that the save method already did it

phase_context is the raw state data passed from state to state...it gets filled in and then save is called to persist it.

Even so, I see that perviously when we rescue the exception, we add it to the phase_context, but then we delete it almost immediately and log it.  Even so, in this else condition we signal provision error and the error caught here is completely lost (at least from what I can tell).